### ISSUE TITLE
Add Caching To Markdown Renderer

### ISSUE THREAD

**Issue by khalidabuhakmeh** [ritterim/releases#19]
I was thinking of using SQL Lite to create a local caching implementation that can survive application reboots. Thoughts?

**Comment by kendaleiv** [comment]
# TL;DR

Markdown caching seems superior to our needs because output cache doesn't seem like a good fit. Also, using durable storage for markdown caching would survive an application restart.

## Output caching vs. markdown renderer caching

After discussion with @khalidabuhakmeh, we determined that output caching might _seem_ like a better approach, but this approach of caching the markdown rendering requests seems ultimately superior.

We want this application to give a live view into application releases. If an individual shares that a new release happens (email, etc.) along with a link to this application -- the information may be out of date. No lengthy caching time may be appropriate in this case.

If we wanted to implement output cache busting, we'd still likely be making requests to GitHub. This may be the same number of requests if we ignore markdown rendering requests.

However, apparently output caching is not currently available in ASP.NET 5, making output caching kind of a non-starter at this time. But, even if it was available, this may be a superior solution for our needs. And, caching the markdown in a durable fashion would survive application restarts, too!

**EDIT:** According to http://weblogs.asp.net/imranbaloch/output-caching-in-aspnet5-mvc6 it looks like output cache was added in ASP.NET 5 beta 3.

**Comment by khalidabuhakmeh** [comment]
@kendaleiv I found this http://www.davepaquette.com/archive/2015/06/03/mvc-6-cache-tag-helper.aspx

**Comment by kendaleiv** [comment]
Cool. I don't think any sort of time based caching will _work_ for our use case -- unless we purge the cache for updates immediately. Even as short as a 5 minute cache time may be too lengthy and lead to confusion / annoyance. And, the expectation of traffic for this app would make 5 minute caching not that useful anyhow.

It may prove useful for future things, though. It looks neat.
